test/splunk_integration_test.go

```diff
@@ -84,3 +92,93 @@ func testVerifyMetricIndexAndSourcetypeAnnotations(t *testing.T) {
 		assert.Greater(t, len(events), 1)
 	})
 }
+
+func testVerifyMetricNamespaceAnnotations(t *testing.T) {
+	namespace := "default"
+	defaultIndex := "ci_metrics"
+	defaultSourcetype := "httpevent"
+	annotationIndex := "test_metrics"
+	annotationSourcetype := "annotation_sourcetype"
+
+	tests := []struct {
+		name                      string
+		annotationIndexValue      string
+		annotationSourcetypeValue string
+	}{
+		{"default index and default sourcetype", "", ""},
+		{"annotation index and default sourcetype", annotationIndex, ""},
+		{"default index and annotation sourcetype", "", annotationSourcetype},
+		{"annotation index and annotation sourcetype", annotationIndex, annotationSourcetype},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			addNamespaceAnnotation(t, namespace, tt.annotationIndexValue, tt.annotationSourcetypeValue)
+			time.Sleep(20 * time.Second)
+
+			// Check metrics are being send to proper index and sourcetype
+			index := defaultIndex
+			if tt.annotationIndexValue != "" {
+				index = tt.annotationIndexValue
+			}
+			sourcetype := defaultSourcetype
+			if tt.annotationSourcetypeValue != "" {
+				sourcetype = tt.annotationSourcetypeValue
+			}
+			searchQuery := METRIC_SEARCH_QUERY_STRING + "index=" + index + " filter=\"sourcetype=" + sourcetype + "\" | search \"k8s.namespace.name\"=" + namespace
+			fmt.Println("Search Query: ", searchQuery)
+			startTime := "-15s@s"
+			events := CheckEventsFromSplunk(searchQuery, startTime)
+			fmt.Println(" =========>  Events received: ", len(events))
+			assert.Greater(t, len(events), 1)
+
+			removeAllNamespaceAnnotations(t, namespace)
+		})
+	}
+}
+
+func removeAllNamespaceAnnotations(t *testing.T, namespace_name string) {
+	testKubeConfig, setKubeConfig := os.LookupEnv("KUBECONFIG")
+	require.True(t, setKubeConfig, "the environment variable KUBECONFIG must be set")
+	kubeConfig, err := clientcmd.BuildConfigFromFlags("", testKubeConfig)
+	require.NoError(t, err)
+	client, err := kubernetes.NewForConfig(kubeConfig)
+	require.NoError(t, err)
```

Comment:
I'd suggest to create the client once in the main test and pass it is input arg to these helper functions